filters/overlayImage.go

```diff
@@ -0,0 +1,206 @@
+package filters
+
+import (
+	"errors"
+	"github.com/zalando-incubator/skrop/parse"
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/h2non/bimg.v1"
+	"io/ioutil"
+	"os"
+)
+
+const (
+	OverlayImageName = "overlayImage"
+	NE               = "NE"
+	NC               = "NC"
+	NW               = "NW"
+	CE               = "CE"
+	CC               = "CC"
+	CW               = "CW"
+	SE               = "SE"
+	SC               = "SC"
+	SW               = "SW"
+)
+
+var (
+	gravityType = map[string]bool{
+		NE: true,
+		NC: true,
+		NW: true,
+		CE: true,
+		CC: true,
+		CW: true,
+		SE: true,
+		SC: true,
+		SW: true,
+	}
+	verticalGravity = map[string]bimg.Gravity{
+		NE: bimg.GravityNorth,
+		NC: bimg.GravityNorth,
+		NW: bimg.GravityNorth,
+		CE: bimg.GravityCentre,
+		CC: bimg.GravityCentre,
+		CW: bimg.GravityCentre,
+		SE: bimg.GravitySouth,
+		SC: bimg.GravitySouth,
+		SW: bimg.GravitySouth,
+	}
+	horizontalGravity = map[string]bimg.Gravity{
+		NE: bimg.GravityEast,
+		NC: bimg.GravityCentre,
+		NW: bimg.GravityWest,
+		CE: bimg.GravityEast,
+		CC: bimg.GravityCentre,
+		CW: bimg.GravityWest,
+		SE: bimg.GravityEast,
+		SC: bimg.GravityCentre,
+		SW: bimg.GravityWest,
+	}
+)
+
+type overlay struct {
+	file              string
+	opacity           float64
+	verticalGravity   bimg.Gravity
+	horizontalGravity bimg.Gravity
+	rightMargin       int
+	leftMargin        int
+	topMargin         int
+	bottomMargin      int
+}
+
+func NewOverlayImage() filters.Spec {
+	return &overlay{}
+}
+
+func (r *overlay) Name() string {
+	return OverlayImageName
+}
+
+func (r *overlay) CreateOptions(image *bimg.Image) (*bimg.Options, error) {
+	origSize, err := image.Size()
+	if err != nil {
+		return nil, err
+	}
+
+	overArr := readImage(r.file)
+	overImage := bimg.NewImage(overArr)
+	overSize, err := overImage.Size()
+	if err != nil {
+		return nil, err
+	}
+
+	var x, y int
+	switch r.verticalGravity {
+	case bimg.GravityNorth:
+		y = r.topMargin
+	case bimg.GravityCentre:
+		y = r.topMargin + int(float64(origSize.Height-r.topMargin-r.bottomMargin)/2) - int(float64(overSize.Height)/2)
```

Comment:
every time I see this in the IDE... I indent the 'case'... and then it is reverted back :(